sale_order_lot_selection/tests/test_sale_order_lot_selection.py

```diff
@@ -1,5 +1,6 @@
 # © 2015 Agile Business Group
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
 import odoo.tests.common as test_common
 
 
```

Comment:
I think now is a good time for a refactoring here.
The setUp steps are executed for each test, I would change an use setUpClass instead, which is executed once for all tests. If you use TransactionCase, the checkpoint is restored after each test.

There's basically a single test: I would split it. By the way, It looks like to me that
It's ok to test a whole process in 1 test, but it's also good to test individual parts of the code.

Also, I have the feeling that this would fix your issues regarding naming of tests methods.
Their order shouldn't matter. I would drop the numbering.
It's more a matter of finding out what are the common needs for setting up the environment, and throw it up in the `setUpClass` method.

Reply:
@nicomacr What do you think of this ? @mmequignon Maybe in another PR ?